src/crab/split_dbm.cpp

```diff
@@ -1008,9 +1008,30 @@ void SplitDBM::apply(arith_binop_t op, variable_t x, variable_t y, const number_
     case arith_binop_t::MUL: assign(x, linear_expression_t(k, y)); break;
     // For the rest of operations, we fall back on intervals.
     case arith_binop_t::SDIV: set(x, get_interval(y, finite_width).SDiv(interval_t(k))); break;
-    case arith_binop_t::UDIV: set(x, get_interval(y, finite_width).UDiv(interval_t(k))); break;
+    case arith_binop_t::UDIV: {
+        if (finite_width == 32) {
+            // As defined in the BPF ISA specification, for 32 bit unsigned division the immediate value is
+            // treated as a 32-bit unsigned integer, so we need to cast it to uint32_t.
+            set(x, get_interval(y, finite_width).UDiv(interval_t(k.cast_to<uint32_t>())));
+        } else {
+            // As defined in the BPF ISA specification, for 64 bit unsigned division the immediate value is
+            // treated as a 32-bit signed integer, so we need to cast it to int32_t.
+            set(x, get_interval(y, finite_width).UDiv(interval_t(k.cast_to<int32_t>())));
+        }
+        break;
```

Comment:
I think it's better to have a dedicated function here. Suggestion (untested):
```cpp
// As defined in the BPF ISA specification, the immediate value of an unsigned modulo is treated differently depending on the width:
// * for 32 bit, as a 32-bit unsigned integer
// * for 64 bit, as a 32-bit (not 64 bit) signed integer
number_t read_imm_for_modulo(const number_t& imm, int width) {
    assert(width == 32 || width == 64);
    if (width == 32) {
        return number_t{k.cast_to<uint32_t>()};
    }
    return number_t{k.cast_to<int32_t>()};
}
```
(of course, feel free to improve the wording, or keep yours).
Note that I intentionally use brace-initialization anywhere possible, since it prevents narrowing.

Reply:
Thanks! I updated it to make the handling of immediate values explicit for sdiv, smod, udiv, and umod for both 32 and 64 bit.